### PR TITLE
Add BECS payment method

### LINKED ISSUE #3621: Add BECS payment method
Stripe integrates BECS as a Bank direct debit and can be integrated into UPE, similar to other payment methods already integrated, like SEPA. More details in the spike: pcqRLn-sM-p2

### PR DESCRIPTION
**Pull request by ismaeldcom**
Fixes #3621 

#### Testing instructions
- WCPay Dev options **Enable grouped settings**, **Enable UPE checkout**, and **Add UPE additional payment methods** must be checked.
- A **Australian** Stripe account is needed for BECS, onboard one if needed.
- Go to **Payments > Settings**.
- **BECS Direct Debit** should be on **Payment methods** list, along with its icon and fees pill _(screenshot 1)_.
- Enable BECS and click **Save changes**, ensure that changes were saved.
- On the store, add an item to your cart, and go to checkout.
- Verify your currency **is not** AUD, BEC should not appear under **Popular payment methods**.
- Switch your currency to AUD, BECS should be available under **Popular payment methods** _(screenshot 2)_.
- Ensure you're running Stripe listener (npm run listen).
- Make 3 purchases using the following accounts numbers and `000000` as BSB:
  - 000123456
  - 111111113
  - 922222227 (3 minutes delay)
- Go to **WooCommerce > Orders**.
- Open the first order (with account number 000123456).
- Click **Refund**, and then increase a line item in your order to 1.
- Click **Refund via WooCommerce Payments**, verify your refund completed.
- Confirm that the orders notes match the ones on the screenshots _(3,4,5)_ below, for the 3 purchases.
- Navigate to **Payments > Transactions**, you should see your **Payment** and **Payment refund** listed for your order, verify both have the **Direct Debit** logo _(screenshot 6)_.
- Click on the Payment, then verify the **Payment method** both at the top and bottom are correctly formatting _(screenshot 7)_.
- Go to **Payments > Settings**, then uncheck **BECS Direct Debit**.
- Verify that the name and the logo in the modal are correct _(screenshot 8)_, then click **Remove** and **Save changes**.
- Verify **BECS Direct Debit** is no longer in enabled the list of **Payment methods**.

#### Screenshots
| 1. Payment methods | 2. Checkout |
|-|-|
|Screenshot 2022-01-19 at 12 44 58|Screenshot 2022-01-19 at 12 46 32|

| 3. Notes 000123456 | 4. Notes 111111113 | 5. Notes 922222227 |
|-|-|-|
|Screenshot 2022-01-19 at 12 32 35|Screenshot 2022-01-19 at 12 34 00|Screenshot 2022-01-19 at 12 34 46|

| 6. Transactions list |
|-|
|Screenshot 2022-01-19 at 12 44 28|

| 7. Transaction details |
|-|
|Screenshot 2022-01-19 at 12 33 26|

| 8. Remove modal |
|-|
|Screenshot 2022-01-19 at 12 45 28|

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: client/payment-details/payment-method/becs/index.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	bsb: '000000',
+	last4: '0000',
+	fingerprint: 'fingerprint placeholder',
+	id: 'id placeholder',
+	name: 'name placeholder',
+	email: 'email placeholder',
+	formattedAddress: 'address placeholder',
+};
+
+const SepaDetails = ( { charge = {}, isLoading } ) => {
```
Comment (jessepearson) [inline client/payment-details/payment-method/becs/index.js:53]: SepaDetails is mentioned here and at the bottom of the file.

**Review by jessepearson — COMMENTED** [review]
Are we going to be adding feature flags for these new payment methods or just relying on enabling/disabling fees on the server as the feature flag? I know this was brought up here: pcqRLn-v7-p2#comment-1061 and the related post.

Also, should https://github.com/Automattic/woocommerce-payments/pull/3683 be merged first?

File: client/payment-details/payment-method/becs/index.js (hunk 1 above)
Comment (ismaeldcom) [reply client/payment-details/payment-method/becs/index.js]: Thanks! I missed it, addressed in 824c9e3

**Comment by ismaeldcom** [comment]
>Are we going to be adding feature flags for these new payment methods or just relying on enabling/disabling fees on the server as the feature flag? I know this was brought up here: pcqRLn-v7-p2#comment-1061 and the related post.

Good question. I'm not sure if we should add a feature flag for them. I personally think that having the option to rely on server fees for that is enough, and doesn't need any extra code that will be removed later.

>Also, should #3683 be merged first?

Yes! Otherwise, it will cause client errors if we remove the fees from some payment methods.

**Review by jessepearson — APPROVED** [review]
Tested according to instructions, and everything is working as expected. 
Thanks again for helping me get webhooks working, or I wouldn't have been able to test this.

I also double checked to make sure #3683 worked in this instance by emptying the array for `COUNTRIES_WITH_BECS` in the server. This made BECS not appear in the list.

Also onboarded as US business and BECS did not appear. When onboarding again with an AU business, it appeared again.

**Pull request merged**